tests/unit/graph/test_selector.py

```diff
@@ -231,350 +217,82 @@ def test_invalid_specs(invalid):
-        inject_plugin(PostgresPlugin)
-        register_adapter(config, get_mp_context())
-        loader = dbt.parser.manifest.ManifestLoader(config, {config.project_name: config})
-        loader.manifest.macros = self.macro_manifest.macros
-        loader.load()
-        return loader.manifest
-
-    def test__single_model(self):
-        self.use_models(
-            {
-                "model_one": "select * from events",
-            }
-        )
-
-        config = self.get_config()
-        manifest = self.load_manifest(config)
+class TestCompiler:
+    def test_single_model(self, runtime_config: RuntimeConfig):
+        model = make_model(pkg="pkg", name="model_one", code="SELECT * FROM events")
+        manifest = make_manifest(nodes=[model])
 
-        compiler = self.get_compiler(config)
+        compiler = dbt.compilation.Compiler(config=runtime_config)
         linker = compiler.compile(manifest)
 
-        self.assertEqual(list(linker.nodes()), ["model.test_models_compile.model_one"])
+        assert linker.nodes() == {model.unique_id}
+        assert linker.edges() == set()
 
-        self.assertEqual(list(linker.edges()), [])
-
-    def test__two_models_simple_ref(self):
-        self.use_models(
-            {
-                "model_one": "select * from events",
-                "model_two": "select * from {{ref('model_one')}}",
-            }
+    def test_two_models_simple_ref(self, runtime_config: RuntimeConfig):
+        model_one = make_model(pkg="pkg", name="model_one", code="SELECT * FROM events")
+        model_two = make_model(
+            pkg="pkg",
+            name="model_two",
+            code="SELECT * FROM {{ref('model_one')}}",
+            refs=[model_one],
         )
+        models = [model_one, model_two]
+        manifest = make_manifest(nodes=models)
 
-        config = self.get_config()
-        manifest = self.load_manifest(config)
-        compiler = self.get_compiler(config)
+        compiler = dbt.compilation.Compiler(config=runtime_config)
         linker = compiler.compile(manifest)
 
-        self.assertCountEqual(
-            linker.nodes(),
-            [
-                "model.test_models_compile.model_one",
-                "model.test_models_compile.model_two",
-            ],
-        )
-
-        self.assertCountEqual(
-            linker.edges(),
-            [
-                (
-                    "model.test_models_compile.model_one",
-                    "model.test_models_compile.model_two",
-                )
-            ],
-        )
-
-    def test__two_models_package_ref(self):
-        self.use_models(
-            {
-                "model_one": "select * from events",
-                "model_two": "select * from {{ref('test_models_compile', 'model_one')}}",
-            }
-        )
+        expected_nodes = [model.unique_id for model in models]
+        assert linker.nodes() == set(expected_nodes)
+        assert list(linker.edges()) == [tuple(expected_nodes)]
```

Comment:
Nitty Nit: It is better to just write out the actual values in the test if it is not too complex. This way reader can understand this edges will yield a list of string without knowing what `model.unique_id`(which is the implementation detail of `model`) is.

Reply:
Would a compromise of adding a type hint be sufficient? For example:
```python
expected_nodes: List[str] = [model.unique_id for model in models]
```

My issue with hand writing it out is that I find that it doesn't scale well and makes things more brittle. From a perspective of making it easier to write tests, I think this approach will suit us better.

Reply:
I should have added this point in the last comment.
So in some way, we are writing out some form of implementation in the tests(it is fairly small in this case). IMHO unit tests should be defining input and output explicitly without (or as little as possible) dependencies between them.
I don't think writing out an explicit value makes writing tests harder.

This is not a very strongly held opinion and I am happy that you make the final call here. We can always adjust later on if we find out more.